### PR TITLE
Adding Passkeys Folder in Smoke and Adding Basic Passkeys Button Veri…

### PR DESCRIPTION
**Pull request by rajendra-deriv**
Adding Passkeys Folder in Smoke and Adding Basic Passkeys Button Verification on Login page

### CODE REVIEW

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST - 142511, QATEST - 125330 - Passkeys Login Page Button Check', () => {
+  const size = ['small', 'desktop']
+
+  beforeEach(() => {
+    Cypress.env('stdConfigAppId', Cypress.env('passkeyAppId'))
```
Comment (markodonnell-deriv) [inline cypress/e2e/smoke/Passkeys/passkeys.cy.js:7]: I wouldn't override the stdConfigAppId as that could affect other tests.

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  })
+
+  size.forEach((size) => {
+    it('Should contain Passkeys Button in Mobile Responsive and Should not contain on Desktop Version', () => {
+      const isMobile = size == 'small' ? true : false
+      cy.c_visitResponsive('/', size)
+      cy.visit(
+        Cypress.env('passkeyUrl') + '?app_id=' + Cypress.env('stdConfigAppId')
```
Comment (markodonnell-deriv) [inline cypress/e2e/smoke/Passkeys/passkeys.cy.js:15]: Just use the passkeyAppId here.

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Some comments

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js (hunk 1 above)
Comment (rajendra-deriv) [reply cypress/e2e/smoke/Passkeys/passkeys.cy.js]: Removed the assignment.

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js (hunk 2 above)
Comment (rajendra-deriv) [reply cypress/e2e/smoke/Passkeys/passkeys.cy.js]: using passkeyAppId directly

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST - 142511, QATEST - 125330 - Passkeys Login Page Button Check', () => {
+  const size = ['small', 'desktop']
+
+  size.forEach((size) => {
+    it('Should contain Passkeys Button in Mobile Responsive and Should not contain on Desktop Version', () => {
+      const isMobile = size == 'small' ? true : false
+      cy.c_visitResponsive('/', size)
+      cy.visit(
```
Comment (markodonnell-deriv) [inline cypress/e2e/smoke/Passkeys/passkeys.cy.js:10]: Put this path in c_visitRepsonsive and you won't need the second visit.
Also, surely you need a login?

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Review by bilalmajid-deriv — APPROVED** [review]

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]
Can you pleas cover all of these steps as well for web version
image

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    it('Should contain Passkeys Button in Mobile Responsive and Should not contain on Desktop Version', () => {
+      const isMobile = size == 'small' ? true : false
+      cy.c_visitResponsive(
+        Cypress.env('passkeyUrl') + '?app_id=' + Cypress.env('passkeyAppId'),
+        size
+      )
+      if (isMobile) {
+        cy.findByText('Passkey').should('be.visible')
+      } else {
+        cy.get('a#btnPasskeys.button.passkey-btn').should('not.exist')
```
Comment (meeyun-deriv) [inline cypress/e2e/smoke/Passkeys/passkeys.cy.js:16]: can we use `findByLabel('Login with passkeys')` or `findByText('Passkey')` for consistency?

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST - 142511, QATEST - 125330 - Passkeys Login Page Button Check', () => {
```
Comment (meeyun-deriv) [inline cypress/e2e/smoke/Passkeys/passkeys.cy.js:3]: Please help to update the test case custom id to `QATEST-142511` and `QATEST-125330` (without spaces before and after the `-`) for easier grep in future

File: cypress.config.js [hunk 6]
```diff
@@ -300,6 +300,8 @@ module.exports = defineConfig({
     actualAmount : process.env.E2E_ACTUAL_AMOUNT,
     appRegisterID: process.env.E2E_APP_REGISTER,
     appRegisterUrl: process.env.E2E_APP_REGISTER_URL,
+    passkeyUrl: process.env.E2E_PASSKEY_URL,
+    passkeyAppId: process.env.E2E_PASSKEY_APP_ID,
```
Comment (meeyun-deriv) [inline cypress.config.js:304]: can you please share the values through LP? so we can add this into GH secret?

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js (hunk 4 above)
Comment (rajendra-deriv) [reply cypress/e2e/smoke/Passkeys/passkeys.cy.js:16]: Updated the code as per comments.

File: cypress/e2e/smoke/Passkeys/passkeys.cy.js (hunk 5 above)
Comment (rajendra-deriv) [reply cypress/e2e/smoke/Passkeys/passkeys.cy.js:3]: Updated as per these changes.

**Comment by rajendra-deriv** [comment]
Hi @bilalmajid-deriv , I have update these steps as per QATEST-125357 test case. The older one needs implementation. So the scope of this test case is good to go.

**Review by bilalmajid-deriv — APPROVED** [review]

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**